src/pathfinding_service/service.py

```diff
@@ -105,7 +105,7 @@ def __init__(  # pylint: disable=too-many-arguments
 
         self.token_networks = self._load_token_networks()
         self.updated = gevent.event.Event()  # set whenever blocks are processed
-        self.startup_finished = gevent.event.Event()
+        self.startup_finished = gevent.event.AsyncResult()
```

Comment:
Why do we need an `AsyncResult` here?

Reply:
Events don't have a `get` that can raise exceptions, so they don't help handling the error case.